charts/chartmuseum/values.yaml

```diff
@@ -0,0 +1,361 @@
+    # oss endpoint to store charts for alibaba storage backend
+    STORAGE_ALIBABA_ENDPOINT:
+    # server side encryption algorithm for alibaba storage backend, can be one
+    # of: AES256 or KMS
+    STORAGE_ALIBABA_SSE:
+    # s3 bucket to store charts for amazon storage backend
+    STORAGE_AMAZON_BUCKET:
+    # prefix to store charts for amazon storage backend
+    STORAGE_AMAZON_PREFIX:
+    # region of s3 bucket to store charts
+    STORAGE_AMAZON_REGION:
+    # alternative s3 endpoint
+    STORAGE_AMAZON_ENDPOINT:
+    # server side encryption algorithm
+    STORAGE_AMAZON_SSE:
+    # gcs bucket to store charts for google storage backend
+    STORAGE_GOOGLE_BUCKET:
+    # prefix to store charts for google storage backend
+    STORAGE_GOOGLE_PREFIX:
+    # container to store charts for microsoft storage backend
+    STORAGE_MICROSOFT_CONTAINER:
+    # prefix to store charts for microsoft storage backend
+    STORAGE_MICROSOFT_PREFIX:
+    # container to store charts for openstack storage backend
+    STORAGE_OPENSTACK_CONTAINER:
+    # prefix to store charts for openstack storage backend
+    STORAGE_OPENSTACK_PREFIX:
+    # region of openstack container
+    STORAGE_OPENSTACK_REGION:
+    # path to a CA cert bundle for your openstack endpoint
+    STORAGE_OPENSTACK_CACERT:
+    # compartment id for for oracle storage backend
+    STORAGE_ORACLE_COMPARTMENTID:
+    # oci bucket to store charts for oracle storage backend
+    STORAGE_ORACLE_BUCKET:
+    # prefix to store charts for oracle storage backend
+    STORAGE_ORACLE_PREFIX:
+    # form field which will be queried for the chart file content
+    CHART_POST_FORM_FIELD_NAME: chart
+    # form field which will be queried for the provenance file content
+    PROV_POST_FORM_FIELD_NAME: prov
+    # levels of nested repos for multitenancy. The default depth is 0 (singletenant server)
+    DEPTH: 0
+    # show debug messages
+    DEBUG: false
+    # output structured logs as json
+    LOG_JSON: true
+    # disable use of index-cache.yaml
+    DISABLE_STATEFILES: false
+    # enable Prometheus metrics
+    ENABLE_METRICS: false
+    # disable all routes prefixed with /api
+    DISABLE_API: true
+    # allow chart versions to be re-uploaded
+    ALLOW_OVERWRITE: false
+    # absolute url for .tgzs in index.yaml
+    CHART_URL:
+    # allow anonymous GET operations when auth is used
+    AUTH_ANONYMOUS_GET: false
+    # sets the base context path
+    CONTEXT_PATH:
+    # parallel scan limit for the repo indexer
+    INDEX_LIMIT: 0
+    # cache store, can be one of: redis (leave blank for inmemory cache)
+    CACHE:
+    # address of Redis service (host:port)
+    CACHE_REDIS_ADDR:
+    # Redis database to be selected after connect
+    CACHE_REDIS_DB: 0
+    # enable bearer auth
+    BEARER_AUTH: false
+    # auth realm used for bearer auth
+    AUTH_REALM:
+    # auth service used for bearer auth
+    AUTH_SERVICE:
+        # username for basic http authentication
```

Comment:
It seems that the lines 99-102 are newly added by us.
According to the document https://github.com/chartmuseum/charts/tree/main/src/chartmuseum#basic-authentication, the authentication information supports mounting through secret, which seems to be safer.

Reply:
I tried to add args on lines 107-109,but secret.yaml cant read them. So I directly wrote the args into the env.open.

Reply:
```bash
# kind create cluster --image=kindest/node:v1.24.13
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.24.13) 🖼
 ✓ Preparing nodes 📦
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Set kubectl context to "kind-kind"
You can now use your cluster with:

kubectl cluster-info --context kind-kind

Thanks for using kind! 😊

# helm repo add chartmuseum https://chartmuseum.github.io/charts

"chartmuseum" has been added to your repositories

# helm install my-chartmuseum chartmuseum/chartmuseum --version 2.15.0

NAME: my-chartmuseum
LAST DEPLOYED: Tue Jul 25 17:54:06 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
** Please be patient while the chart is being deployed **

Get the ChartMuseum URL by running:

  export POD_NAME=$(kubectl get pods --namespace default -l "app=chartmuseum" -l "release=my-chartmuseum" -o jsonpath="{.items[0].metadata.name}")
  echo http://127.0.0.1:8080/
  kubectl port-forward $POD_NAME 8080:8080 --namespace default

# export POD_NAME=$(kubectl get pods --namespace default -l "app=chartmuseum" -l "release=my-chartmuseum" -o jsonpath="{.items[0].metadata.name}")
# echo http://127.0.0.1:8080/
# kubectl port-forward $POD_NAME 8080:8080 --namespace default

Forwarding from 127.0.0.1:8080 -> 8080
Forwarding from [::1]:8080 -> 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080
Handling connection for 8080

# Then visit http://127.0.0.1:8080/, will show chartmuseum start page.
# curl http://127.0.0.1:8080/                
<!DOCTYPE html>
<html>
<head>
<title>Welcome to ChartMuseum!</title>
<style>
    body {
        width: 35em;
        margin: 0 auto;
        font-family: Tahoma, Verdana, Arial, sans-serif;
    }
</style>
</head>
<body>
<h1>Welcome to ChartMuseum!</h1>
<p>If you see this page, the ChartMuseum web server is successfully installed and
working.</p>

<p>For online documentation and support please refer to the
<a href="https://github.com/helm/chartmuseum">GitHub project</a>.<br/>

<p><em>Thank you for using ChartMuseum.</em></p>
</body>
</html>
        %
```
Then let's try basic auth
```bash
# helm uninstall my-chartmuseum 
release "my-chartmuseum" uninstalled

# helm install my-chartmuseum chartmuseum/chartmuseum --version 2.15.0 --set env.secret.BASIC_AUTH_USER=user-name --set env.secret.BASIC_AUTH_PASS=pass-word
NAME: my-chartmuseum
LAST DEPLOYED: Tue Jul 25 18:04:19 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
** Please be patient while the chart is being deployed **

Get the ChartMuseum URL by running:

  export POD_NAME=$(kubectl get pods --namespace default -l "app=chartmuseum" -l "release=my-chartmuseum" -o jsonpath="{.items[0].metadata.name}")
  echo http://127.0.0.1:8080/
  kubectl port-forward $POD_NAME 8080:8080 --namespace default

# one secret will store our username and password.
# kubectl get secret my-chartmuseum-chartmuseum -o jsonpath='{.data}'

{"BASIC_AUTH_PASS":"cGFzcy13b3Jk","BASIC_AUTH_USER":"dXNlci1uYW1l"}%


# export POD_NAME=$(kubectl get pods --namespace default -l "app=chartmuseum" -l "release=my-chartmuseum" -o jsonpath="{.items[0].metadata.name}")
# echo http://127.0.0.1:8080/
# kubectl port-forward $POD_NAME 8080:8080 --namespace default

http://127.0.0.1:8080/
Forwarding from 127.0.0.1:8080 -> 8080
Forwarding from [::1]:8080 -> 8080
Handling connection for 8080
Handling connection for 8080

# Then visit chartmuseum start page will need input username and password, we should input `user-name` and `pass-word`
# curl http://127.0.0.1:8080/                                       
{"error":"unauthorized"}

# curl -u user-name:pass-word http://127.0.0.1:8080/                
<!DOCTYPE html>
<html>
<head>
<title>Welcome to ChartMuseum!</title>
<style>
    body {
        width: 35em;
        margin: 0 auto;
        font-family: Tahoma, Verdana, Arial, sans-serif;
    }
</style>
</head>
<body>
<h1>Welcome to ChartMuseum!</h1>
<p>If you see this page, the ChartMuseum web server is successfully installed and
working.</p>

<p>For online documentation and support please refer to the
<a href="https://github.com/helm/chartmuseum">GitHub project</a>.<br/>

<p><em>Thank you for using ChartMuseum.</em></p>
</body>
</html>
        %

```